Description: Improve hg-git compatibility
 Recent versions of Mercurial check if a filectx is the same as the
 parents, and avoid creating a new filelog. This is what we want, but
 hg-git doesn't do that.
Author: Felipe Contreras <felipe.contreras@gmail.com>
Last-Update: 2016-09-19
Commit: b81ec14c2e84cb4b1e0d8223d3821fa735813b80
Date: Mon, 16 May 2016 22:08:57 -0500

---
 git-remote-hg | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/git-remote-hg b/git-remote-hg
index a0d9086..17c4c85 100755
--- a/git-remote-hg
+++ b/git-remote-hg
@@ -813,7 +813,18 @@ def parse_commit(parser):
             else:
                 raise IOError
         if 'ctx' in of:
-            return of['ctx']
+            if mode == 'hg':
+                ctx = of['ctx']
+                is_exec = ctx.isexec()
+                is_link = ctx.islink()
+                if check_version(3, 1):
+                    return context.memfilectx(repo, f, ctx.data(),
+                            is_link, is_exec)
+                else:
+                    return context.memfilectx(f, ctx.data(),
+                            is_link, is_exec)
+            else:
+                return of['ctx']
         is_exec = of['mode'] == 'x'
         is_link = of['mode'] == 'l'
         rename = of.get('rename', None)
